### PR TITLE
add goreleaser example

### LINKED ISSUE #165: goreleaser example
One of the use cases of using SLSA generators is generating artifacts with GoReleaser and using a generic generator to generate provenance, so, it'd be good to add an example of this to ensure that everything is working as we expected during new releases of both GoReleaser and generators. Maybe we could add draft-release support to test the new behavior https://github.com/slsa-framework/slsa-github-generator/issues/1476.

/cc @caarlos0

WDYT @ianlewis @laurentsimon @asraa ?

### PR DESCRIPTION
**Pull request by developer-guy**
fixes #165 

I tried my best but there is no doubt that it needs improvement this is why I marked this PR as a WIP.

### CODE REVIEW

File: .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+" # # triggers only if push new tag version, like `v0.8.4` or else
+
+permissions: read-all
+
+env:
+  PAT_TOKEN: ${{ secrets.E2E_GENERIC_TOKEN }}
+  GH_TOKEN: ${{ github.token }}
+  ISSUE_REPOSITORY: slsa-framework/slsa-github-generator
+  DEFAULT_VERSION: v18.0.0
```
Comment (laurentsimon) [inline .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml:15]: make sure to pick a unique default version. Check if this one is already used in another workflow.

File: .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      - id: checkout
+        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+        with:
+          fetch-depth: 0
+      - id: ldflags
+        run: |
+          set -euo pipefail
+
+          THIS_FILE=$(gh api -H "Accept: application/vnd.github.v3+json" "/repos/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" | jq -r '.path' | cut -d '/' -f3)
+          BRANCH=$(echo "$THIS_FILE" | cut -d '.' -f4)
```
Comment (laurentsimon) [inline .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml:54]: are these needed or they were only needed for the Go builder?

**Review by laurentsimon — COMMENTED** [review]
added a few comments, but I think it looks good.

**Comment by ianlewis** [comment]
> I tried my best but there is no doubt that it needs improvement this is why I marked this PR as a WIP.

No worries! All PRs are a WIP in reality 😄

File: .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml (hunk 2 above)
Comment (developer-guy) [reply .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml:54]: I'm using the same Go code in GoReleaser as well, this is why I remained them as is.

File: .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/e2e.generic.tag.main.goreleaser.slsa3.yml:15]: thanks

**Comment by developer-guy** [comment]
kindly ping @laurentsimon

**Pull request merged**